src/realm/array_flex.hpp

```diff
@@ -0,0 +1,45 @@
+/*************************************************************************
+ *
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **************************************************************************/
+
+#ifndef REALM_ARRAY_FLEX_HPP
+#define REALM_ARRAY_FLEX_HPP
+
+#include <realm/array_encode.hpp>
+
+namespace realm {
+//
+// Compress array in Flex format
+// Decompress array in WTypeBits formats
+//
+class ArrayFlex : public ArrayEncode {
```

Comment:
Array flex is the compression implementor. It is a member of Array. We should consider move into this class all the encoded related implementation, I guess we may want to implement ad hoc "move", "remove" methods for our compressed array. Profiling a little bit the code here and there, and decode is the most run function, every time we do some operation on the array.